java/rocksjni/sst_file_readerjni.cc

```diff
@@ -71,6 +71,19 @@ jlong Java_org_rocksdb_SstFileReader_newIterator(JNIEnv * /*env*/,
   return GET_CPLUSPLUS_POINTER(sst_file_reader->NewIterator(*read_options));
 }
 
+/*
+ * Class:     org_rocksdb_SstFileReader
+ * Method:    newTableIterator
+ * Signature: (JJ)J
+ */
+jlong Java_org_rocksdb_SstFileReader_newTableIterator(JNIEnv * /*env*/,
+                                                      jclass /*jcls*/,
+                                                      jlong jhandle) {
+  auto *sst_file_reader =
+      reinterpret_cast<ROCKSDB_NAMESPACE::SstFileReader *>(jhandle);
+  return GET_CPLUSPLUS_POINTER(sst_file_reader->NewTableIterator().release());
```

Comment:
I am a bit worried about the `release()` here? What is the ownership model for the iterator in your code please?

Reply:
It is the same as sstFileReaderIterator. So release should be triggered manually irrespective of sstFileReader class